src/bot/snoopiBenchBot.jl

```diff
@@ -0,0 +1,132 @@
+################################################################
+"""
+    timesum(snoop)
+
+Calculates and prints the total time measured by a snoop macro. It is used inside @snoopiBenchBot.
+
+Julia can cache inference results so to measure the effect of adding _precompile_() sentences generated by snoopi to your package, use the [`@snoopiBenchBot`](@ref). This benchmark measures inference time taken during loading and running of a package.
+
+
+## Manually benchmarking withtout using [`@snoopiBenchBot`](@ref)
+- dev your package
+
+- comment the precompile part of your package (`include()` and `_precompile_()`)
+- run the following benchmark
+- restart Julia
+
+- uncomment the precompile part of your package (`include()` and `_precompile_()`)
+- run the following benchmark
+- restart Julia
+
+### Benchmark
+```julia
+using SnoopCompile
+
+println("Package load time:")
+loadSnoop = @snoopi using MatLang
+
+timesum(loadSnoop)
+
+println("Running Examples/Tests:")
+runSnoop = @snoopi begin
+    using MatLang
+    include(joinpath(dirname(dirname(pathof(MatLang))),"test","runtests.jl"))
+end
+
+timesum(runSnoop)
+```
+"""
+function timesum(snoop::Vector{Tuple{Float64, Core.MethodInstance}})
+    if isempty(snoop)
+        return 0.0
+    else
+        return sum(first, snoop)
+    end
+end
+
+################################################################
+"""
+    @snoopiBenchBot(packageName::String, snoopScript::Expr)
+    @snoopiBenchBot(packageName::String)
+
+Performs an infertime benchmark by activating and deactivating the __precompile__()
+# Examples
+Benchmarking the load infer time
+```julia
+println("loading infer benchmark")
+
+@snoopiBenchBot "MatLang" using MatLang
+```
+
+Benchmarking the example infer time
+```julia
+println("examples infer benchmark")
+
+@snoopiBenchBot "MatLang" begin
+    using MatLang
+    examplePath = joinpath(dirname(dirname(pathof(MatLang))), "examples")
+    # include(joinpath(examplePath,"Language_Fundamentals", "usage_Entering_Commands.jl"))
```

Comment:
What's the purpose of having this commented line?

Reply:
I needed the path of the test package so I can cd to it, but without loading the package. So I decided to have an external Julia process.